api/tests/core/search/test_serialize_algolia.py

```diff
@@ -219,6 +219,46 @@ def test_serialize_collective_offer():
     }
 
 
+def test_serialize_collective_offer_without_institution():
+    domain1 = educational_factories.EducationalDomainFactory(name="Danse")
+    domain2 = educational_factories.EducationalDomainFactory(name="Architecture")
+    collective_offer = educational_factories.CollectiveOfferFactory(
+        dateCreated=datetime.datetime(2022, 1, 1, 10, 0, 0),
+        name="Titre formidable",
+        students=[StudentLevels.CAP1, StudentLevels.CAP2],
+        subcategoryId=subcategories.LIVRE_PAPIER.id,
+        venue__postalCode="86140",
+        venue__name="La Moyenne Librairie SA",
+        venue__publicName="La Moyenne Librairie",
+        venue__managingOfferer__name="Les Librairies Associées",
+        venue__departementCode="86",
+        educational_domains=[domain1, domain2],
+    )
+
+    serialized = algolia.AlgoliaBackend().serialize_collective_offer(collective_offer)
+    assert serialized == {
+        "objectID": collective_offer.id,
+        "offer": {
+            "dateCreated": 1641031200.0,
+            "name": "Titre formidable",
+            "students": ["CAP - 1re année", "CAP - 2e année"],
+            "subcategoryId": subcategories.LIVRE_PAPIER.id,
+            "domains": [domain1.id, domain2.id],
+            "educationalInstitutionUAICode": "all",
+        },
+        "offerer": {
+            "name": "Les Librairies Associées",
+        },
+        "venue": {
+            "departmentCode": "86",
+            "id": collective_offer.venue.id,
+            "name": "La Moyenne Librairie SA",
+            "publicName": "La Moyenne Librairie",
+        },
+        "isTemplate": False,
+    }
```

Comment:
(détail / question de goût) Si je comprends bien, on cherche ici à vérifier la valeur d'une seule clé de l'objet sérialisé. Je proposerais de simplifier drastiquement l'initialisation et les vérifications pour que ce soit plus clair (et qu'on n'ait pas besoin de mettre à jour ce test si on rajoute un autre champ à la sérialisation) : 
```suggestion
def test_serialize_collective_offer_without_institution():
    collective_offer = educational_factories.CollectiveOfferFactory(institution=None)
    serialized = algolia.AlgoliaBackend().serialize_collective_offer(collective_offer)
    assert serialized["offer"]"educationalInstitutionUAICode"] = "all"
```

Reply:
c'est vrai vu qu'on a un test au dessus qui vérifie le reste, on peut simplifier :)